docs/v2.md

```diff
@@ -0,0 +1,163 @@
+majority of changes backwards compatible. In fact, for Javascript users, the changes
+are completely backwards compatible for Avro schemas.
+
+For Typescript users, however, or if you are migrating from using Avro to using
+one of the other schema types, there are a few things to keep in mind.
+
+The main change is that the core `Schema` type has changed from an Avro specific
+schema to a generic schema interface for all schema types. The old schema looked
+like this:
+
+```ts
+interface AvroSchema {
+  name: string
+  namespace?: string
+  type: 'record'
+  fields: any[]
+  toBuffer(payload: object): Buffer
+  fromBuffer(buffer: Buffer, resolver?: Resolver, noCheck?: boolean): any
+  isValid(
+    payload: object,
+    opts?: { errorHook: (path: Array<string>, value: any, type?: any) => void },
+  ): boolean
+}
+```
+
+This is still the case for Avro schemas, but for Protobuf or Json Schema, a
+more generic schema type is used:
+
+```ts
+interface Schema {
+  toBuffer(payload: object): Buffer
+  fromBuffer(buffer: Buffer, resolver?: Resolver, noCheck?: boolean): any
+  isValid(
+    payload: object,
+    opts?: { errorHook: (path: Array<string>, value: any, type?: any) => void },
+  ): boolean
+}
+```
+
+If you have code that uses any of the Avro specific fields on the schema
+(for example returned by `getSchema`), you may need to first narrow the type
+to `AvroSchema`:
+
+```ts
+import { AvroSchema, Schema } from '@kafkajs/confluent-schema-registry'
+
+function isAvroSchema(schema: AvroSchema | Schema): schema is AvroSchema {
+  return (schema as AvroSchema).name != null
+}
+
+const schema = await registry.getSchema(registryId)
+
+if (isAvroSchema(schema)) {
+  // schema is now `AvroSchema`
+  const { name, namespace, type, fields } = schema
+}
+```
+
+## Adapting to new APIs
+
+This is **entirely optional**, as all the old APIs should be retained (with the above caveat
+about the schema type), but if you want to adapt your code to the new API to make
+future migrations easier, these are the changes you would need to make.
+
+### Configuring serialization libraries
+
+In version 1, the client constructor took a second `options` argument with
+a single option `forSchemaOptions` that was passed directly to
+[`avsc.Type.forSchema`](https://github.com/mtth/avsc/wiki/API#typeforschemaschema-opts)
+as the `opts` argument.
+
+```js
+const { SchemaRegistry } = require('@kafkajs/confluent-schema-registry')
+
+const registry = new SchemaRegistry(
+  { host: 'http://localhost:8081' },
+  { forSchemaOptions: { noAnonymousTypes: true }}
+)
+```
+
+Since we now support multiple schema types, these options have been moved
+one level into a schema type specific option:
+
+```js
+const { SchemaRegistry, SchemaType } = require('@kafkajs/confluent-schema-registry')
+
+const registry = new SchemaRegistry(
+  { host: 'http://localhost:8081' },
+  {
+    [SchemaType.AVRO]: { forSchemaOptions: { noAnonymousTypes: true }},
```

Comment:
I think this should be `[SchemaType.AVRO]: { noAnonymousTypes: true }` (remove the `forSchemaOptions`)

Reply:
Fixed in c5b3134 👍